DEVELOPMENT.md

```diff
@@ -0,0 +1,536 @@
+
+## Installation Options
+
+### Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+- **Python 3.8+** - [Download Python](https://www.python.org/downloads/)
+- **Poetry** - Python dependency manager
+  ```bash
+  curl -sSL https://install.python-poetry.org | python3 -
+  ```
+- **MongoDB** - Either installed locally or via Docker
+  ```bash
+  # Using Docker
+  docker run -d -p 27017:27017 --name mongodb mongo:latest
+
+  # Or install MongoDB Community Edition
+  # See: https://docs.mongodb.com/manual/installation/
+  ```
+- **Node.js & npm** (for frontend) - [Download Node.js](https://nodejs.org/)
+- **Docker & Docker Compose** (optional, for full stack) - [Download Docker](https://www.docker.com/get-started)
+
+## Running the Frontend
+
+### Using nyrkio_frontend.py (Recommended)
+
+The `etc/nyrkio_frontend.py` script manages the frontend development server with support for different deployment modes:
+
+```bash
+# Start frontend with local backend (default)
+python3 etc/nyrkio_frontend.py start
+
+# Start frontend with production backend
+python3 etc/nyrkio_frontend.py start --mode production
+
+# Start on custom port
+python3 etc/nyrkio_frontend.py start --port 3000
+
+# Check frontend status
+python3 etc/nyrkio_frontend.py status
+
+# Stop frontend
+python3 etc/nyrkio_frontend.py stop
+
+# Restart frontend
+python3 etc/nyrkio_frontend.py restart
+
+# Install dependencies only
+python3 etc/nyrkio_frontend.py install
+```
+
+**Deployment Modes:**
+
+- **local** (default): Uses `vite.config.test.js` - proxies API requests to `http://localhost:8001`
+- **production**: Uses `vite.config.js` - proxies API requests to `https://nyrk.io`
+
+**Features:**
+- Auto-detects and installs npm dependencies
+- Runs server in background with PID tracking
+- Configurable port (default: 5173)
+- Graceful shutdown support
+
+**Frontend Configuration Files:**
+
+- `frontend/vite.config.js` - Production configuration (proxies to https://nyrk.io)
+- `frontend/vite.config.test.js` - Local development configuration (proxies to http://localhost:8001)
+
+### Using npm Directly
+
+For foreground execution with full output:
+
+```bash
+cd frontend
+
+# Install dependencies (first time only)
+npm install
+
+# Start with local config
+npm run dev -- --config vite.config.test.js --port 5173
+
+# Start with production config
+npm run dev -- --config vite.config.js --port 5173
+```
+
+**Default URLs:**
+- Frontend: http://localhost:5173
+- API (local): http://localhost:8001
+- API (production): https://nyrk.io
+
+### Full Stack Development
+
+For a complete local development environment:
+
+```bash
+# Terminal 1: Start backend
+python3 etc/nyrkio_backend.py start
+
+# Terminal 2: Start frontend in local mode
+python3 etc/nyrkio_frontend.py start
+
+# Or run both and check status
+python3 etc/nyrkio_backend.py status
+python3 etc/nyrkio_frontend.py status
+```
+
+## Running the Backend
```

Comment:
Likewise this is very welcome to work on the backend without docket <3

Where is the DB in this setup? (Document it, don't reply)